### ISSUE TITLE
OSX Python 3 wheels

### ISSUE THREAD

**Issue by amadain17** [ParallelSSH/parallel-ssh#128]
On a mac (High Sierra v 10.13.5) I successfully installed parallel-ssh, pssh, ssh2 using pip3. When I tried to import ParallelSSHClient I got a Symbol not found: _libssh2_channel_request_auth_agent

1. IEC02R5A6DG8WP:pythonScripts mdev0002$ python3
Python 3.6.4 (default, Mar 16 2018, 09:27:59) 
[GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from pssh.clients import ParallelSSHClient
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.6/site-packages/pssh/clients/__init__.py", line 19, in <module>
    from .native.parallel import ParallelSSHClient
  File "/usr/local/lib/python3.6/site-packages/pssh/clients/native/__init__.py", line 19, in <module>
    from .parallel import ParallelSSHClient
  File "/usr/local/lib/python3.6/site-packages/pssh/clients/native/parallel.py", line 25, in <module>
    from .single import SSHClient
  File "/usr/local/lib/python3.6/site-packages/pssh/clients/native/single.py", line 33, in <module>
    from ssh2.session import Session
  File "ssh2/channel.pxd", line 24, in init ssh2.session
ImportError: dlopen(/usr/local/lib/python3.6/site-packages/ssh2/channel.cpython-36m-darwin.so, 2): Symbol not found: _libssh2_channel_request_auth_agent
  Referenced from: /usr/local/lib/python3.6/site-packages/ssh2/channel.cpython-36m-darwin.so
  Expected in: flat namespace
 in /usr/local/lib/python3.6/site-packages/ssh2/channel.cpython-36m-darwin.so
>>>

**Comment by lordzouga** [comment]
I have this same problem. Were you able to find a workaround @amadain17 ?

**Comment by pkittenis** [comment]
Hi there,

Thanks for the interest and the report. 

Could you please check if there is a libssh2 package installed on the system? Perhaps installed via homebrew or ports?

To confirm, this is with parallel-ssh version `1.8.0`, installed via binary wheel right? Specifically, [this wheel](https://files.pythonhosted.org/packages/cb/dc/7bbefe7eac890505fda978fdc373a4fc4baea9227bb2da49f4807b2f0760/parallel_ssh-1.8.0-cp27-cp27m-macosx_10_13_x86_64.whl)?

**Comment by pkittenis** [comment]
Note - as workaround can install previous version with `pip install parallel-ssh==1.7.0`

**Comment by pkittenis** [comment]
Ah, just noticed `python3` being used. The wheels provided for OSX are for python2 only.

Will need to follow [install from source instructions](http://ssh2-python.readthedocs.io/en/latest/installation.html#installation-from-source) for python 3 for now.

**Comment by micmarty** [comment]
@pkittenis Building from source seems not to work (but maybe just in my case). 
**Anyone succeeded?**

**Comment by pkittenis** [comment]
[1.8.0.post1](https://github.com/ParallelSSH/parallel-ssh/releases/tag/1.8.0.post1) has added Python 3.6 and 3.7 wheels for OSX, as well as Linux 3.7. 

That will unblock those of you running Python 3 on OSX.

At the moment would recommend sticking to 3.6 as gevent has some crashing issues on 3.7 which are also blocking windows 3.7 wheels.

Source builds would better automatically build and use the embedded libssh2 by default so they get the correct libssh2 version, but that will be a separate change.

Any issues with source builds please feel free to create a new issue.

**Comment by tkrasnoperov** [comment]
We're using `parallel-ssh-1.8.1` and `ssh2-python-0.15.0.post9` on MacOSX 10.13.3. Wheel is `ssh2_python-0.15.0.post9-cp35-cp35m-macosx_10_6_x86_64.whl`.

```
$ brew info libssh2
pythlibssh2: stable 1.8.0 (bottled), HEAD
C library implementing the SSH2 protocol
https://libssh2.org/
o/usr/local/Cellar/libssh2/1.8.0 (182 files, 797.8KB) *
  Poured from bottle on 2018-09-07 at 15:02:25
From: https://github.com/Homebrew/homebrew-core/blob/master/Formula/libssh2.rb
==> Dependencies
Required: openssl ✔
==> Options
--HEAD
	Install HEAD version
```
```
$ python --version
Python 3.5.2 :: Anaconda custom (x86_64)
```

And we get:
```
$ python -c "from pssh.clients import ParallelSSHClient"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/anaconda3/envs/apm/lib/python3.5/site-packages/pssh/clients/__init__.py", line 19, in <module>
    from .native.parallel import ParallelSSHClient
  File "/anaconda3/envs/apm/lib/python3.5/site-packages/pssh/clients/native/__init__.py", line 19, in <module>
    from .parallel import ParallelSSHClient
  File "/anaconda3/envs/apm/lib/python3.5/site-packages/pssh/clients/native/parallel.py", line 25, in <module>
    from .single import SSHClient
  File "/anaconda3/envs/apm/lib/python3.5/site-packages/pssh/clients/native/single.py", line 33, in <module>
    from ssh2.session import Session
  File "ssh2/channel.pxd", line 24, in init ssh2.session
ImportError: dlopen(/anaconda3/envs/apm/lib/python3.5/site-packages/ssh2/channel.cpython-35m-darwin.so, 2): Symbol not found: _libssh2_channel_request_auth_agent
  Referenced from: /anaconda3/envs/apm/lib/python3.5/site-packages/ssh2/channel.cpython-35m-darwin.so
  Expected in: flat namespace
 in /anaconda3/envs/apm/lib/python3.5/site-packages/ssh2/channel.cpython-35m-darwin.so
```

**Comment by romperstomper** [comment]
Switching to python3.6 fixed it for me when I ran into this issue with python3.4.4 on OSX High Sierra:

ImportError: dlopen(/Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/ssh2/channel.so, 2): Symbol not found: _libssh2_channel_request_auth_agent
  Referenced from: /Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/ssh2/channel.so
  Expected in: flat namespace
 in /Library/Frameworks/Python.framework/Versions/3.4/lib/python3.4/site-packages/ssh2/channel.so

**Comment by pkittenis** [comment]
Hi @tkrasnoperov ,

Can you try after running `brew uninstall libssh2`? It looks like the system libssh2 is overriding the embedded libssh2 in the wheel which is a different version.
